### PR TITLE
Reorganize files

### LINKED ISSUE #7: Reorganize build_scripts files
bin/release_build.rb
config/*
lib/rpm_build.rb
lib/rpm_build/{clone_repos.rb, generate_gemset.rb, etc.}
scripts/{npm_registry, gem_cleanup.sh}

Also move classes to its own file and rename some classes/files as it's pretty confusing right now...

### PR DESCRIPTION
**Pull request by simaishi**
Fixes #7 

Moved files to bin/config/lib, etc. as appropriate. Ruby scripts are now in ManageIQ:RPMBuild namespace.

Marking as WIP as I still need to rename classes/files. Currently:

- ManageIQ::RPMBuild::SourceReps(clone repos and copy rpm spec files to build root)
- ManageIQ::RPMBuild::GemSet (generate `gemset` rpm contents, by running bundle install, update:ui, etc.)
- ManageIQ::RPMBuild::ManageIQTarBuild (generate 'manageiq' rpm contents)
- ManageIQ::RPMBuild::MakeTarFile (create 3 tarballs)
- ManageIQ::RPMBuild::RpmBuild (kick off rpm build, currently works with Copr only)

Any suggestion is appreciated.

### CODE REVIEW

File: lib/manageiq/rpm_build/generate_gemset.rb [hunk 1]
```diff
@@ -0,0 +1,141 @@
+require 'awesome_spawn'
+require 'fileutils'
+require 'pathname'
+require 'yaml'
+
+module ManageIQ
+  module RPMBuild
+    class GemSet
```
Comment (Fryguy) [inline lib/manageiq/rpm_build/generate_gemset.rb:8]: Minor, but the class name should match the path name (so either rename the file or rename the class.

Comment (Fryguy) [reply lib/manageiq/rpm_build/generate_gemset.rb:8]: (Oh just saw you already noted this in the OP)

**Comment by simaishi** [comment]
How about this?

**ManageIQ::RPMBuild::SetupSourceRepos** = clone repos and copy rpm spec files to build root
**ManageIQ::RPMBuild::GenerateGemSet** = generate gemset rpm contents, by running bundle install, update:ui, etc.
**ManageIQ::RPMBuild::GenerateCore? ::GenerateManageIQ? ::GenerateVMDB?** = generate 'manageiq' rpm contents (compile assets, SUI build, etc.)
**ManageIQ::RPMBuild::MakeTarFiles** = create 3 tarballs
**ManageIQ::RPMBuild::BuildCoprRPM** = kick off rpm build (I can extract common code when we're adding support for non-Copr)

**Comment by Fryguy** [comment]
`ManageIQ::RPMBuild::GenerateCore` seems good

**Comment by simaishi** [comment]
Classes/files renamed.

**Pull request merged**